parsers/forth.c

```diff
@@ -0,0 +1,78 @@
+/*
+*   Copyright (c) 2023, Eric Forgeot
+*
+*   Based on work by Jon Strait
+*
+*   This source code is released for free distribution under the terms of the
+*   GNU General Public License version 2 or (at your opinion) any later version.
+*
+*   This module contains functions for generating tags for Forth files
+*/
+
+/*
+*   INCLUDE FILES
+*/
+#include "general.h"	/* must always come first */
+
+#include <ctype.h>
+#include <string.h>
+
+#include "parse.h"
+#include "read.h"
+#include "vstring.h"
+#include "routines.h"
+#include "entry.h"
+
+/*
+*   DATA DEFINITIONS
+*/
+
+typedef enum {
+//   K_SECTION,
+   K_FUNCTION,
+} ForthKind;
+
+static kindDefinition ForthKinds[] = {
+	{ true, 's', "word", "words" },
+};
+
+/*
+*   FUNCTION DEFINITIONS
+*/
+
+static void findForthTags (void)
+{
+	vString *name = vStringNew();
+	const unsigned char *line;
+
+	while ((line = readLineFromInputFile()) != NULL)
+	{
+		if (line[0] == ':') {
+			vStringCatS(name, " ");
```

Comment:
Do you really want to put a white space at the beginning of the tag name?

If the whitespace is for decorating the name rendered on your IDE or editor, it should be done on the IDE or editor; it is not the job of ctags.

Reply:
It doesn't seem necessary, I've removed it